File: example/petstore-retrofit2/src/main/kotlin/com/example/petstore/client/Pet.kt
package com.example.petstore.client

import java.math.* // ktlint-disable
import java.util.* // ktlint-disable
import java.io.* // ktlint-disable
import com.squareup.moshi.* // ktlint-disable

/**
 * 
 *
 * 
 * @link https://github.com/eaglesakura/swagger-codegen-extensions
 **/
@Suppress("unused")
data class Pet(

    /**
     * 
     */
    @Json(name = "id")
    val id: kotlin.Long? = null, 

    /**
     * 
     */
    @Json(name = "category")
    val category: Category? = null, 

    /**
     * 
     */
    @Json(name = "name")
    val name: kotlin.String, 

    /**
     * 
     */
    @Json(name = "photoUrls")
    val photoUrls: kotlin.Array<kotlin.String>, 

    /**
     * 
     */
    @Json(name = "tags")
    val tags: kotlin.Array<Tag>? = null, 

    /**
     * pet status in the store
     */
    @Json(name = "status")
    val status: PetStatusEnum? = null
) {

    /**
     * Convert to Json.
     */
    fun toJson(): String =
        InternalUtils.moshi.adapter(Pet::class.java).toJson(this)

    companion object {
        /**
         * Try parse json.
         */
        fun parse(json: String): Pet? {
            return InternalUtils.moshi.adapter(Pet::class.java).fromJson(json)
        }
    }
}


data class PetStatusEnum(private val value: String) {
    override fun toString(): String {
        return value
    }
    companion object {

        val available = PetStatusEnum("available")

        val pending = PetStatusEnum("pending")

        val sold = PetStatusEnum("sold")

        val MoshiTypeAdapter = object : JsonAdapter<PetStatusEnum>() {
            @Throws(IOException::class)
            override fun fromJson(reader: JsonReader): PetStatusEnum {
                return parse(reader.nextString())
            }

            @Throws(IOException::class)
            override fun toJson(writer: JsonWriter, value: PetStatusEnum?) {
                if(value == null) {
                    return
                }
                writer.value(value.toString())
            }

            override fun toString(): String {
                return "JsonAdapter(PetStatusEnum)"
            }
        }
        fun parse(value: String): PetStatusEnum {
            return when(value) {
                "" -> throw IllegalArgumentException("Invalid enum($value)")
                
                "available" -> available
                
                "pending" -> pending
                
                "sold" -> sold
                
                else -> PetStatusEnum(value)
            }
        }
    }
}

File: example/petstore-dart2/lib/model/pet.dart
part of client_petstore.api;

class Pet {
  
  int id = null;
  

  Category category = null;
  

  String name = null;
  

  List<String> photoUrls = [];
  

  List<Tag> tags = [];
  
/* pet status in the store */
  String status = null;
  //enum statusEnum {  available,  pending,  sold,  };
  Pet();

  @override
  String toString() {
    return 'Pet[id=$id, category=$category, name=$name, photoUrls=$photoUrls, tags=$tags, status=$status, ]';
  }

  Pet.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
      if(json.containsKey("id")) {
              id =
                      json['id']
              ;
      }
      if(json.containsKey("category")) {
              category =
                  
                      
                      new Category.fromJson(json['category'])
;
      }
      if(json.containsKey("name")) {
              name =
                      json['name']
              ;
      }
      if(json.containsKey("photoUrls")) {
              photoUrls =
                      (json['photoUrls'] as List).map((item) => item as String).toList()
              ;
      }
      if(json.containsKey("tags")) {
              tags =
                  Tag.listFromJson(json['tags'])
;
      }
      if(json.containsKey("status")) {
              status =
                      json['status']
              ;
      }
  }


  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'category': category,
      'name': name,
      'photoUrls': photoUrls,
      'tags': tags,
      'status': status
     };
  }

  static List<Pet> listFromJson(List<dynamic> json) {
    return json == null ? new List<Pet>() : json.map((value) => new Pet.fromJson(value)).toList();
  }

  static Map<String, Pet> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, Pet>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new Pet.fromJson(value));
    }
    return map;
  }
}

